### DISCUSSION TITLE
Does ArgoCD wait for resources marked with 'PreSync' to become fully healthy before proceeding to the sync phase?

### DISCUSSION THREAD

**Discussion by alita1991** [argoproj/argo-helm#2941] [Q&A] [answered]
Hi,

Based on my testing, ArgoCD waits for resources labeled as 'PreSync' to reach a healthy state (for example, I have a DaemonSet in the pre-sync phase). However, I haven't been able to find any mention of this behavior in the documentation, except for jobs.

This is necessary for a feature where I need a DaemonSet to be deployed first and reach a healthy state before deploying other resources.

Can anyone provide documentation or guidance on this?

Thanks

**Comment by yu-croco** [comment, marked as answer]
Hi @alita1991 , since it's about Argo CD itself, you can ask at [upstream](https://github.com/argoproj/argo-cd).
*We provide Helm Chart for Argoproj but we don't handle the specific feature of them.
